### DISCUSSION TITLE
ToggleableType in EntityFinder

### DISCUSSION THREAD

**Discussion by Jan-DT** [Minestom/Minestom#2091] [Help]
Just a quick question, what does the `ToggleableType` in the EntityFinder methods mean?
```java
public EntityFinder setGameMode(@NotNull GameMode gameMode, @NotNull @NotNull ToggleableType toggleableType) {
    this.gameModes.put(gameMode, toggleableType.getValue());
    return this;
}
```
It is also used in the `EntityFinder#setName`, `EntityFinder#setUuid` and `EntityFinder#setEntity` methods.
If I just want to set the gamemode of every player in the EntityFinder, what do I set this to?

**Comment by Tofaa2** [comment]
Try something like this

```java
EntityFinder ef = context.get(entityArgorWhatever)

ef.find(commandSender)
    .stream().filter(e -> e instanceof Player)
    .forEach(e -> ((Player)e).setGameMode(GameMode.CREATIVE));
```

**Reply by Jan-DT** [reply]
That was my backup plan, though I am wondering what these methods would otherwise be used for.
